src/util/psbt/map/output.rs

```diff
@@ -92,6 +92,13 @@ impl PartialEq for TapTree {
 
 impl Eq for TapTree {}
 
+impl AsRef<TaprootBuilder> for TapTree {
```

Comment:
nit: 

I would rather let users call `as_builder` instead rather than an indirect `AsRef`. Maybe it's just only me, but `TapTree` and `TaprootBuilder` are not similar conceptually in the operations they do. I would only conservatively implement `AsRef` when the conversion is super-obvious or should be hidden from the user.

Reply:
`AsRef` is needed for generic function arguments

Reply:
I don't like leaking the internal representation at all. Is there a strong reason to do it?

Reply:
Yes, otherwise we can't use `TapTree` in places where `TaprootBuilder` is required as an argument: `fn some_stuff(tree: impl AsRef<TaprootBuilder>)`

Reply:
As mentioned in the above comment, I would rather users use conversion functions.